Pipeline/SimplePackageDeploy/SimplePackageDeploy.groovy

```diff
@@ -0,0 +1,276 @@
+ *     --workDir option.
+ *  4) Untars the deployment package specified by --tarFileName to the temporary 
+ *     folder created in the previous step.
+ *  5) Reads the BuildReport.json file from the temporary folder.
+ *  6) Identifies all the components to be deployed from the BuildReport.json file.
+ *  7) Identifies deploy types for the components to be deployed from the 
+ *     BuildReport.json file.
+ *  8) Identifies the target libraries based on the deploy type by concatenating the
+ *     --hlq value with the correspoding LLQ value from the JSON string - targetLibLLQMap.
+ *  9) Identifies the copy mode based on the deploy type and corresponding copy mode 
+ *     from the JSON string - copyModeMap.
+ * 10) Copies the components using the copy mode to the respective target libraries.
+ * 11) Deletes the temporary package extract folder after deployment.
+ *
+ * RETURNS:
+ *    Exit Status 0       -  Success
+ *    Exit Status 1       -  Failure
+ *
+ ************************************************************************************/
+
+@Field Properties props = null
+props = parseInput(args)
+
+def startTime = new Date()
+props.startTime = startTime.format("yyyyMMdd.hhmmss.mmm")
+println("** SimplePackageDeploy start at $props.startTime")
+
+// Map of last level dataset qualifier to DBB CopyToPDS CopyMode.
+def copyModeMap = parseJSONStringToMap(props.copyModeMap)
+// Map of last level dataset qualifier to Target PDS last level qualifier.
+def targetLibLLQMap = parseJSONStringToMap(props.targetLibLLQMap)
+
+// Define and create tar file extract directory 
+def tarExtractDirName = "${props.workDir}/DeployFiles_${props.startTime}"
+def tarExtractDir = new File("${tarExtractDirName}")
+
+!tarExtractDir.exists() ?: tarExtractDir.deleteDir()
+tarExtractDir.mkdirs()
+println("** Created tar file extract directory $tarExtractDirName")       
+
+// Check if the tar file provided is existing 
+def tarFile = new File("${props.tarFileName}")
+
+if(!tarFile.exists()){
+    println("** Tar file not found at $tarFile")
+    System.exit(1)
+}
+
+
+// Untar the tar file
+println("** Untar file at $tarFile.")
+
+def processCmd = ["sh", "-c", "tar -C $tarExtractDirName -xvf $tarFile"]
+
+def rc = runProcess(processCmd)
+if (rc == 0) println("Package untar done to $tarExtractDirName")
+else {
+	println("Failed to untar $tarFile")
+	System.exit(1)
+}
+
+
+// Check if BuildReport.json exists, deploy files if exists and if not stop deploy 
+
+def buildReportFile = new File("${tarExtractDirName}/BuildReport.json")
+if (buildReportFile.exists()) {	
+	deployFromBuildReport(buildReportFile,tarExtractDirName,targetLibLLQMap,copyModeMap)
+} else {
+	println("** Build report data at $tarExtractDirName/BuildReport.json not found")
+	println("** Deployment stopped")
+}
+
+if (tarExtractDir.exists()) tarExtractDir.deleteDir()
+println("\nCleaning up the temporary folder - ${tarExtractDirName}\n")
```

Comment:
Please prefix with the `**` to have a better reading of the output log. Also, I don't think we need the newline at the beginning.

Reply:
Done